eng/native/configurecompiler.cmake

```diff
@@ -18,7 +20,7 @@ include(${CMAKE_CURRENT_LIST_DIR}/configureoptimization.cmake)
 #-----------------------------------------------------
 
 if(MSVC)
-    add_compile_options(/Zi /FC /Zc:strictStrings)
```

Comment:
What happened to `/Zc:strictStrings` ?

Reply:
It was causing an error in installer:

```cmd
.\build.cmd -subsetcategory installer

           fxr_resolver.cpp
       6>C:\Users\adeel\Source\Repos\runtime2\src\installer\corehost\cli\hostmisc\pal.windows.cpp(230,51): error C2440: 'initializing': cannot convert from 'const unsigned short [18]' to 'pal::char_t *' [C:\Users\adeel\Source\Repos\runtime2\artifacts\obj\win-x64.Debug\corehost\cli\hostmisc\libhostmisc.vcxproj]
         C:\Users\adeel\Source\Repos\runtime2\src\installer\corehost\cli\hostmisc\pal.windows.cpp(230,26): message : Conversion from string literal loses const qualifier (see /Zc:strictStrings) [C:\Users\adeel\Source\Repos\runtime2\artifacts\obj\win-x64.Debug\corehost\cli\hostmisc\libhostmisc.vcxproj]
           longfile.windows.cpp
       6>C:\Users\adeel\Source\Repos\runtime2\src\installer\corehost\cli\hostmisc\pal.windows.cpp(280,52): error C2440: '=': cannot convert from 'const unsigned short [18]' to 'pal::char_t *' [C:\Users\adeel\Source\Repos\runtime2\artifacts\obj\win-x64.Debug\corehost\cli\hostmisc\libhostmisc.vcxproj]
         C:\Users\adeel\Source\Repos\runtime2\src\installer\corehost\cli\hostmisc\pal.windows.cpp(280,29): message : Conversion from string literal loses const qualifier (see /Zc:strictStrings) [C:\Users\adeel\Source\Repos\runtime2\artifacts\obj\win-x64.Debug\corehost\cli\hostmisc\libhostmisc.vcxproj]
```

I forgot to add it in CoreCLR, will do. Or maybe I should first try to make installer compatible with `strictStrings`. 💭

Reply:
Yes, it would be nice to sprinkle `const` as necessary to fix this.

Reply:
I am running a clean Windows build on side, will get to this point soon.

Reply:
Using `const` in few places fixed the build error with`strictStrings`:  4fc15c2. Thanks. 👍